### PR TITLE
Lagrange zk Coprocessor CLI Tooling

### PR DESCRIPTION
**Pull request by solipsis**
See avs-smart-contract repo for complete test of registration flow

### CODE REVIEW

File: src/avs/lagrangeZK/lagrangeZK.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	"github.com/etherfi-protocol/etherfi-avs-operator-tool/src/etherfi"
+	"github.com/etherfi-protocol/etherfi-avs-operator-tool/src/gnosis"
+	"github.com/etherfi-protocol/etherfi-avs-operator-tool/src/utils"
+)
+
+// API handle for all core lagrangeZK functionality
+type API struct {
+	Client *ethclient.Client
+
+	ServiceManager            *ZKMRServiceManager
```
Comment (jtfirek) [inline src/avs/lagrangeZK/lagrangeZK.go:21]: Are you just including this to keep the APIs consistent? Doesn't seem like we ever interact with the service manager directly just use the address as a param for the `OperatorAVSRegistrationDigestHash`

File: src/avs/lagrangeZK/lagrangeZK.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		ServiceManagerAddress:     cfg.LagrangeZKMRServiceManagerAddress,
+		StakeRegistry:             stakeRegistry,
+		StakeRegistryAddress:      cfg.LagrangeZKMRStakeRegistryAddress,
+		AvsOperatorManagerAddress: cfg.AvsOperatorManagerAddress,
+		EigenlayerAPI:             eigenlayer.New(cfg, rpcClient),
+	}
+}
+
+// PrepareRegistration aggregates all required info from the node operator that
+// the ether.fi admin will need to register them to the Witness Chain AVS
```
Comment (jtfirek) [inline src/avs/lagrangeZK/lagrangeZK.go]: Witness Chain -> LagrangeZK

File: bin/avs-cli/lagrangeZK/register.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		},
+	},
+}
+
+func handleRegister(ctx context.Context, cli *cli.Command) error {
+
+	// parse cli params
+	inputFilepath := cli.String("registration-input")
+
+	// read input file with required eigenDA data
```
Comment (jtfirek) [inline bin/avs-cli/lagrangeZK/register.go]: eigenDA -> LagrangeZK

**Review by jtfirek — APPROVED** [review]

File: bin/avs-cli/lagrangeZK/register.go (hunk 3 above)
Comment (solipsis) [reply bin/avs-cli/lagrangeZK/register.go]: ty

File: src/avs/lagrangeZK/lagrangeZK.go (hunk 2 above)
Comment (solipsis) [reply src/avs/lagrangeZK/lagrangeZK.go]: ty

File: src/avs/lagrangeZK/lagrangeZK.go (hunk 1 above)
Comment (solipsis) [reply src/avs/lagrangeZK/lagrangeZK.go:21]: yeah. Doesn't need to be there but I think long term its likely there are some calls we would need to support such as changingQuorums and such. Doesn't matter too much either way

**Pull request merged**